### PR TITLE
Add doc section about secured Kafka

### PR DESCRIPTION
**Pull request by rdettai**
### Description

It is common to used an externally managed Kafka service. In that case authentication is necessary, and it would be good to show that we support it.

### How was this PR tested?

Ran tests using aiven.io and the Quickwit Docker distribution.

### CODE REVIEW

**Comment by rdettai** [comment]
@fmassot should we recommend one of SSL or SASL? I'm a bit bothered by the fact that the SASL password ends up being stored clear text in the metastore. I feel SSL is a bit better in that regard because the secrets stay on the Quickwit instances.

File: docs/ingest-data/kafka.md [hunk 1]
```diff
@@ -176,6 +176,58 @@ curl -XPOST -H 'Content-Type: application/json' 'http://localhost:7280/api/v1/gh
 }'
 ```
 
+
+## Secured Kafka connection (optional)
+
+The Quickwit Kafka source supports SSL and SASL authentication. This is
+particularly useful when consuming an external Kafka service.
```
Comment (guilload) [inline docs/ingest-data/kafka.md:183]:
```suggestion
particularly useful when consuming data from an external Kafka service.
```

File: docs/ingest-data/kafka.md [hunk 2]
```diff
@@ -176,6 +176,58 @@ curl -XPOST -H 'Content-Type: application/json' 'http://localhost:7280/api/v1/gh
...
+source_type: kafka
+num_pipelines: 2
+params:
+  topic: gh-archive
+  client_params:
+    bootstrap.servers: "your-kafka-broker.com"
+    security.protocol: "SSL"
+    ssl.ca.location: "/path/to/ca.pem"
+    ssl.certificate.location: "/path/to/service.cert"
+    ssl.key.location: "/path/to/service.key"
```
Comment (guilload) [inline docs/ingest-data/kafka.md:199]:
```suggestion
    bootstrap.servers: your-kafka-broker.com
    security.protocol: SSL
    ssl.ca.location: /path/to/ca.pem
    ssl.certificate.location: /path/to/service.cert
    ssl.key.location: /path/to/service.key
```

File: docs/ingest-data/kafka.md [hunk 3]
```diff
@@ -176,6 +176,58 @@ curl -XPOST -H 'Content-Type: application/json' 'http://localhost:7280/api/v1/gh
...
+num_pipelines: 2
+params:
+  topic: gh-archive
+  client_params:
+    bootstrap.servers: "your-kafka-broker.com"
+    ssl.ca.location: "/path/to/ca.pem"
+    security.protocol: "SASL_SSL"
+    sasl.mechanisms: "SCRAM-SHA-256"
+    sasl.username: "your_sasl_username"
+    sasl.password: "your_sasl_password"
```
Comment (guilload) [inline docs/ingest-data/kafka.md:217]:
```suggestion
  client_params:
    bootstrap.servers: your-kafka-broker.com
    ssl.ca.location: /path/to/ca.pem
    security.protocol: SASL_SSL
    sasl.mechanisms: SCRAM-SHA-256
    sasl.username: your_sasl_username
    sasl.password: your_sasl_password
```

**Review by guilload — APPROVED** [review]
Looking great!

**Pull request merged**